native/src/generalRouter.cpp

```diff
@@ -433,8 +433,28 @@ double GeneralRouter::evaluateCache(RouteDataObjectAttribute attr, const SHARED_
 	return res;
 }
 
+void GeneralRouter::putCache(RouteDataObjectAttribute attr, const SHARED_PTR<RoutingIndex>& reg, std::vector<uint32_t>& types, double val, bool extra) {
+	if (USE_CACHE) {
+		auto regCacheIt = cacheEval[(unsigned int)attr].find(reg);
+		MAP_INTV_DOUBLE* regCache;
+		if (regCacheIt == cacheEval[(unsigned int)attr].end()) {
+			cacheEval[(unsigned int)attr][reg] = MAP_INTV_DOUBLE();
+			regCache = &cacheEval[(unsigned int)attr][reg];
+		} else {
+			regCache = &regCacheIt->second;
+		}
+		types.push_back(extra ? 1 : 0);
+		(*regCache)[types] = val;
+		types.pop_back();
+	}
+}
+
 bool GeneralRouter::acceptLine(const SHARED_PTR<RouteDataObject>& way) {
-	int res = (int)evaluateCache(RouteDataObjectAttribute::ACCESS, way, 0);
+	double res = evaluateCache(RouteDataObjectAttribute::ACCESS, way, 0);
+	if (res < 0) {
```

Comment:
code is different. if (res == null ) in Java

This code couldn't fix bug cause it's a speedup / cache

Reply:
Looks like this code is present in evaluateCache